scuba/config.py

```diff
@@ -103,15 +123,17 @@ def from_yaml(self, node):
             raise yaml.YAMLError(f"Key {key!r} not found in {filename}")
         return cur
 
-    def override(self, node):
+    def override(self, node: yaml.nodes.Node) -> OverrideMixin:
         """
         Implements !override constructor
         """
         # Load the content from the node, as a scalar
+        assert isinstance(node, yaml.nodes.ScalarNode)
         content = self.construct_scalar(node)
+        assert isinstance(content, str)
 
         # Dynamically add an OverrideMixin to the resulting object's type
-        obj = yaml.load(content, lambda s: Loader(s, root=self._root))
+        obj = yaml.load(content, self._rooted_loader(root=self._root))
```

Comment:
cc @adamjseitz: Just FYI since you implemented this in #177

Reply:
Thanks for the ping. Nice to see type annotations - and nice to see them doing their job, as it looks like it caught my previous mistake here.

Reply:
I'm not sure I'd call it a mistake as much as I would an overly-specific type annotation.

Typeshed [says](https://github.com/python/typeshed/blob/d74bea5e870708cb88a0ae8ca6f3e13ca5098531/stubs/PyYAML/yaml/__init__.pyi#L38) that the `Loader` arg is a `type[_Loader | _CLoader]`. But it is only [called](https://github.com/yaml/pyyaml/blob/957ae4d495cf8fcb5475c6c2f1bce801096b68a5/lib/yaml/__init__.py#L79) to create a loader instance. So I could argue that it should be relaxed and annotated as `Callable[[TextIO], _Loader | _CLoader]`, which your code would have satisfied.

Anyway, thanks for taking a look!